bazelfe-core/src/bep_junit/bep_junit_app.rs

```diff
@@ -78,25 +77,22 @@ async fn main() -> Result<(), Box<dyn Error>> {
     let mut failed_actions = Vec::default();
     let mut aborted_actions = Vec::default();
     let mut failed_tests = Vec::default();
-    for build_event in res.iter() {
+    for build_event in hydrated_infos.into_iter() {
         match build_event {
-            bazelfe_bazel_wrapper::bep::build_events::hydrated_stream::HydratedInfo::BazelAbort(abort_info) => {
+            HydratedInfo::BazelAbort(abort_info) => {
                 emit_junit_xml_from_aborted_action(
-                    abort_info,
+                    &abort_info,
                     aborted_actions.len(),
                     &opt.junit_output_path,
                 );
-                aborted_actions.push(abort_info.label.clone());
+                aborted_actions.push(abort_info.label);
             }
-            bazelfe_bazel_wrapper::bep::build_events::hydrated_stream::HydratedInfo::ActionFailed(
-                action_failed,
-            ) => {
-                emit_junit_xml_from_failed_action(action_failed, &opt.junit_output_path);
-                failed_actions.push(action_failed.label.clone());
+            HydratedInfo::ActionFailed(action_failed) => {
+                emit_junit_xml_from_failed_action(&action_failed, &opt.junit_output_path);
+                failed_actions.push(action_failed.label);
             }
-            bazelfe_bazel_wrapper::bep::build_events::hydrated_stream::HydratedInfo::Progress(_) => (),
-            bazelfe_bazel_wrapper::bep::build_events::hydrated_stream::HydratedInfo::TestResult(r) => {
-                if let bazelfe_bazel_wrapper::bep::build_events::build_event_server::bazel_event::TestStatus::Failed =  r.test_summary_event.test_status {
+            HydratedInfo::TestResult(r) => {
+                if r.test_summary_event.test_status.didnt_pass() {
```

Comment:
this is the main change, the rest is avoiding some clones and reducing duplication.